src/components/molecules/ExportProgressItem.tsx

```diff
@@ -0,0 +1,615 @@
+      for (const folderPath of keys(folderMap)) {
+        if (isExportingCanceled()) {
+          return
+        }
+
+        if (folderPath == '/' && !exportSettings.exportingStorage) {
+          continue
+        }
+
+        if (
+          !exportSettings.exportingStorage &&
+          ((exportSettings.recursive &&
+            !folderPath.startsWith(exportSettings.folderPathname)) ||
+            (!exportSettings.recursive &&
+              folderPath != exportSettings.folderPathname))
+        ) {
+          continue
+        }
+
+        const fullPath = join(
+          rootDir,
+          folderPath.substring(folderPath.indexOf(exportSettings.folderName))
+        )
+
+        updateExportState({
+          exportSubOperationTitle: `Creating directory: ${fullPath.replace(
+            rootDir,
+            '.'
+          )}`,
+        })
+        await mkdir(fullPath)
+      }
+    },
+    [
+      exportSettings.exportingStorage,
+      exportSettings.folderName,
+      exportSettings.folderPathname,
+      exportSettings.recursive,
+      isExportingCanceled,
+      updateExportState,
+    ]
+  )
+
+  const convertValueToPercentage = useCallback(
+    (value: number, maxValue: number, exportProgressMaxValue = 100) => {
+      if (maxValue === 0) {
+        return 0
+      }
+      return Math.floor((value / maxValue) * exportProgressMaxValue)
+    },
+    []
+  )
+
+  const calculateNumberOfExportingNotes = useCallback(
+    (noteMap: ObjectMap<NoteDoc>): number => {
+      const calculateNoteCountCondition = !(
+        exportSettings.recursive && noteCount !== 0
+      )
+      let exportingNoteCount = calculateNoteCountCondition ? 0 : noteCount
+      if (exportSettings.exportingStorage) {
+        exportingNoteCount = values(noteMap).length
+      } else {
+        if (calculateNoteCountCondition || exportingNoteCount === 0) {
+          for (const [, noteDoc] of entries(noteMap)) {
+            if (
+              noteDoc.trashed ||
+              (exportSettings.recursive &&
+                !noteDoc.folderPathname.startsWith(
+                  exportSettings.folderPathname
+                )) ||
+              (!exportSettings.recursive &&
+                noteDoc.folderPathname != exportSettings.folderPathname)
+            ) {
+              continue
+            }
+            exportingNoteCount++
+          }
+        }
+      }
+
+      return exportingNoteCount
+    },
+    [
+      exportSettings.exportingStorage,
+      exportSettings.folderPathname,
+      exportSettings.recursive,
+      noteCount,
+    ]
+  )
+
+  const exportNotes = useCallback(
+    async (
+      noteMap: ObjectMap<NoteDoc>,
+      attachmentMap: ObjectMap<Attachment>,
+      rootDir: string
+    ): Promise<number> => {
+      const totalNotesToExport = calculateNumberOfExportingNotes(noteMap)
+
+      let exportingNoteIndex = 0
+      for (const [, noteDoc] of entries(noteMap)) {
```

Comment:
We've already known what to export while executing `calculateNumberOfExportingNotes`. I think we should refactor `calculateNumberOfExportingNotes` so exporting doesn't have to check whole notes twice.

Reply:
Refactored this, now looks better, no repetition!